drivers/modem/ublox-sara-n310.c

```diff
@@ -155,6 +164,52 @@ static int modem_atoi(const char *s, const int err_value, const char *desc,
 	return ret;
 }
 
+/* send AT command */
+static int send_at_command(struct modem_iface *iface,
+			   struct modem_cmd_handler *handler,
+			   const struct modem_cmd *handler_cmds,
+			   size_t handler_cmds_len, const uint8_t *buf,
+			   struct k_sem *sem, k_timeout_t timeout, bool tx_lock)
+{
+	int ret;
+
+	/* wake module if asleep */
+	if (is_awake() == 0) {
+		ret = turn_on_module();
+
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	if (tx_lock) {
+		ret = modem_cmd_send(iface, handler, handler_cmds,
+				     handler_cmds_len, buf, sem, timeout);
+	} else {
+		ret = modem_cmd_send_nolock(iface, handler, handler_cmds,
+					    handler_cmds_len, buf, sem,
+					    timeout);
+	}
+
+	return ret;
+}
+
+/* network state getter */
+int n310_get_state(void)
```

Comment:
maybe change function name to something more explicit (what state is it getting?)